cpp/src/parquet/stream_writer.h

```diff
@@ -0,0 +1,135 @@
+#ifndef PARQUET_STREAM_WRITER_H
+#define PARQUET_STREAM_WRITER_H
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parquet/api/writer.h"
+
+namespace parquet {
+
+// A class for writing Parquet files using an output stream type API.
+//
+// The values given must be of the correct type i.e. the type must
+// match the file schema exactly otherwise a ParquetException will be
+// thrown.
+//
+// The user must explicitly advance to the next row using the
+// nextRow() function.
+//
+// A maximum row group size can be configured, the default size is 512
+// Mio.  Alternatively the row group size can be set to zero and the
+// user can create new row groups by calling the newRowGroup()
+// function.
+//
+// Currently there is no support for optional fields.
+//
+class PARQUET_EXPORT StreamWriter {
+ public:
+  explicit StreamWriter(std::unique_ptr<ParquetFileWriter>& File_Writer);
+
+  ~StreamWriter();
+
+  static void SetDefaultMaxRowGroupSize(const std::int64_t max_size);
+
+  void SetMaxRowGroupSize(const std::int64_t max_size);
+
+  // Moving is possible.
+  StreamWriter(StreamWriter&&) noexcept = default;
+  StreamWriter& operator=(StreamWriter&&) noexcept = default;
+
+  // Copying is not allowed.
+  StreamWriter(const StreamWriter&) = delete;
+  StreamWriter& operator=(const StreamWriter&) = delete;
+
+  StreamWriter& operator<<(const bool v);
+
+  StreamWriter& operator<<(const std::int8_t v);
+
+  StreamWriter& operator<<(const std::uint8_t v);
+
+  StreamWriter& operator<<(const std::int16_t v);
+
+  StreamWriter& operator<<(const std::uint16_t v);
+
+  StreamWriter& operator<<(const std::int32_t v);
+
+  StreamWriter& operator<<(const std::uint32_t v);
+
+  StreamWriter& operator<<(const std::int64_t v);
+
+  StreamWriter& operator<<(const std::uint64_t v);
+
+  StreamWriter& operator<<(const std::chrono::milliseconds& v);
+
+  StreamWriter& operator<<(const std::chrono::microseconds& v);
+
+  StreamWriter& operator<<(const float v);
+
+  StreamWriter& operator<<(const double v);
+
+  StreamWriter& operator<<(const char v);
+
+  template <int N>
+  StreamWriter& operator<<(const char (&v)[N]) {
+    return writeFixedLength_(v, N);
+  }
+
+  StreamWriter& operator<<(const char* v);
+
+  StreamWriter& operator<<(const std::string& v);
```

Comment:
it might pay to make an override for arrow::string_view as well, but this might make circular dependencies worse.

Reply:
Ok I will add arrow::util::string_view as an output type to the StreamWriter class.

Reply:
Actually I think it might not work with constness, sorry for bad suggestion

Reply:
I do not see any issue with constness.